### ISSUE TITLE
Automatization of compatibility tests on sonarqube versions on a specific plugin version

### ISSUE THREAD

**Issue by dedece35** [green-code-initiative/ecoCode#257]
Make an automatic process (with Github Actions) to check if a plugin version (new or old) is compatible on different versions of SonarQube.
It must be based on our current docker files.

**Comment by lbg-oju** [comment]
Hello 👋, can I be assigned to this issue?

**Comment by lbg-oju** [comment]
Is there also a compatibility matrix or similar for this somewhere?

**Comment by glalloue** [comment]
Hi @lbg-oju Sure you can take this issue

Do you know how to do ?  Do you need more information ? 
@dedece35 is the technical referent on this subject if you need help

**Comment by lbg-oju** [comment]
Yeah, it would be great to have more info, eg not sure what’s meant with the last sentence ”based on our current docker file”. My understanding so far is that something like below needs to be done in the cicd?

https://tech.osteel.me/posts/a-github-workflow-to-check-the-compatibility-of-your-php-package-with-a-range-of-dependency-versions

https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#

I hope @dedece35 can confirm that I’m on the right track?

**Comment by dedece35** [comment]
Hi @lbg-oju,
ok thank you for helping !

the last sentence ”based on our current docker file” means that each ecocode plugin contains a `Dockerfile` and a `docker-compose.yml` for local developments by a contributor.
I mean that we can imagine a system based on these files, why not ?

Currently, the different SonarQube compatibilities are mentioned on `README.md` file inside each ecocode plugin repository.
But why not to have a centralized matrix used for your validation system and referenced in each `README.md` file.

If you have some ideas, please, feel free to share and discuss about it.
